core/msgio/android.go

```diff
@@ -44,6 +46,8 @@ func SyncAndroidChannel(ctx context.Context, fc FCMClient, channel *models.Chann
 	start := time.Now()
 
 	if _, err := fc.Send(ctx, sync); err != nil {
+		VerifyTokenIDs(ctx, rt, channel, fcmID)
```

Comment:
The new endpoint is only used for releasing a channel no? So why are we worried about verifying and updating the channel's FCM ID ?

Reply:
I think we need to verify the token any time, we get an error, the same way we have that currently

https://github.com/nyaruka/rapidpro/blob/eef6b0ae6e4a0c73d272cd7d321cd96deab63068/temba/channels/android/sync.py#L118

Reply:
We should probably be verifying ids for channels not being deleted, but for this PR, which is just adding an endpoint to trigger a sync, which for now is only used for deleting channels.. why do you need to check and update the channel being deleted?

Reply:
This was not added in the past as mailroom triggered sync for MT messages only, 
Now that we'll move all the FCM sync in mailroom I think we need the support the verification

Reply:
Unless I misunderstood the goal here is be able to change 

https://github.com/nyaruka/rapidpro/pull/5363/files#diff-91191a502d6ace5d683f8adfba0f12843f4e875eecb6daf7e5a5c4656458324cL684-R685

Reply:
So even MO messages will start calling mailroom to make the FM sync and then we need to support the verification here

Reply:
Ok I didn't see we also trigger a sync when a channel is claimed so the new endpoint isn't just for when we delete a channel.. that's why I was wondering why we need to check tokens for a deleted channel.

Should we be clearing out the token for any error from `VerifyIDToken` or should we be calling `auth.IsIDTokenRevoked(err) || auth.IsUserDisabled(err)`? Seems like that method call could fail for other reasons that don't necessarily mean the token is invalid.

Reply:
Or should we be checking the error from `Send`?

Reply:
Any event use the method `SyncAndroidChannel` to trigger FCM sync

We also only do the verification when the FCM send returned an error

Reply:
Added `auth.IsIDTokenRevoked(err) || auth.IsUserDisabled(err)`